server/sonar-server/src/main/java/org/sonar/server/component/ComponentCleanerService.java

```diff
@@ -51,11 +51,17 @@ public void delete(DbSession dbSession, List<ComponentDto> projects) {
     }
   }
 
+  public void deleteBranch(DbSession dbSession, ComponentDto branch) {
+    // TODO: detect if other branches depend on it?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)